### PR TITLE
[RNMobile][Embed block] Add the top 5 specific embed blocks to the Block inserter list

### PR DESCRIPTION
**Pull request by fluiddot**
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3995

## Description

Add the top 5 specific embed blocks to the Block inserter list ~~and use the colored version of the icons~~:
- YouTube
- Twitter
- WordPress
- ~~Instagram~~ (**NOTE: This block is not added yet, it will when we tackle [this issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3951)**)
- Vimeo

**EDIT:** As mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/34967#issuecomment-930381158), the colored versions of the icons won't be used.

## How has this been tested?

1. Open a post/page.
2. Tap on the ➕ button to open the block inserter list.
3. Scroll down the block list.
4. Observe that the previously mentioned providers have their own block in the list.

## Screenshots 

**iOS:**

Light mode|Dark mode
--|--
|

**Android:**

Light mode|Dark mode
--|--
|

## Types of changes

New feature

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-editor/src/components/block-icon/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-icon/index.native.js:46]: `color` style property is not supported in the `View` component ([reference](https://reactnative.dev/docs/0.64/view-style-props)) so this line wasn't really applying the foreground color defined in the icon.

File: packages/block-editor/src/components/block-icon/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-icon/index.native.js:40]: I noticed we're passing a spread style object as component props, in this case, since the `iconPlaceholder` and `iconPlaceholderDark` styles only define the `fill` property, it matches the expected props for the `Icon` component. However, I have some concerns about this approach since it could lead to unexpected errors if the styles contain other props. For this reason, I decided to refactor it and include this color, which is the default color, in the calculation of the fill color of the icon.

File: packages/block-editor/src/components/block-icon/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-icon/index.native.js:34]: I decided to apply a small refactor for calculating the fill color, now it includes the foreground icon (more info in [this comment](https://github.com/WordPress/gutenberg/pull/34967/files#r712868424)) and the default color (more info in [this comment](https://github.com/WordPress/gutenberg/pull/34967/files#r712874556)).

File: packages/block-editor/src/components/inserter/block-types-tab.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/inserter/block-types-tab.native.js]: The Instagram block won't be available in the block picker until we include [its variation from Jetpack](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/extensions/extended-blocks/core-embed/instagram.js).

File: packages/components/src/mobile/inserter-button/style.native.scss
Comment (fluiddot) [inline packages/components/src/mobile/inserter-button/style.native.scss:40]: We no longer require to define the fill color here because now that we're using the `BlockIcon` component for rendering the icon, these colors are already used as the default color:
https://github.com/WordPress/gutenberg/blob/55bf89b1c84aa5a1ddd17dc92909fdae4cc11a73/packages/block-editor/src/components/block-icon/style.native.scss#L1-L7

File: packages/components/src/mobile/inserter-button/index.native.js
Comment (fluiddot) [inline packages/components/src/mobile/inserter-button/index.native.js:45]: `modalIconStyle` no longer has a dark mode variant because the fill color is now set by default in the `BlockIcon` component (related to [this comment](https://github.com/WordPress/gutenberg/pull/34967/files#r712886224)).

File: packages/components/src/mobile/inserter-button/index.native.js
Comment (fluiddot) [inline packages/components/src/mobile/inserter-button/index.native.js]: Initially, I thought to set `showColors` to `true` for all cases as we have in the web version:
https://github.com/WordPress/gutenberg/blob/44062c586af16dcb0c7ae376f38d2de16b2ed81c/packages/block-editor/src/components/inserter-list-item/index.js#L135

But I realized that Jetpack blocks are also showing the colored version (see attached screenshot), so I finally decided to only enable this for the embed blocks.



This is not happening in the web version because Jetpack determines the icon color depending on the site type ([referece](https://github.com/Automattic/jetpack/blob/3b090d0c81c218a103637aacea4600919e069cd6/projects/plugins/jetpack/extensions/shared/block-icons.js#L13-L29)), unfortunately, this logic is not working in the native version because it relies on a property that is only exposed in web ([reference](https://github.com/Automattic/jetpack/blob/3b090d0c81c218a103637aacea4600919e069cd6/projects/plugins/jetpack/extensions/shared/site-type-utils.js#L11-L38)).

File: packages/block-editor/src/components/block-icon/index.native.js
Comment (jd-alexander) [reply packages/block-editor/src/components/block-icon/index.native.js:46]: Makes sense. Good eye here @fluiddot 👍🏾

File: packages/block-editor/src/components/inserter/block-types-tab.native.js
Comment (jd-alexander) [reply packages/block-editor/src/components/inserter/block-types-tab.native.js]: Thanks for noting this.

File: packages/components/src/mobile/inserter-button/index.native.js
Comment (jd-alexander) [reply packages/components/src/mobile/inserter-button/index.native.js]: Good observation. If we ever need any enhancements here for other icons we can update this logic in the future.

**Review by jd-alexander — APPROVED** [review]
Thanks for these changes @fluiddot I reviewed the code and I learnt a lot from the comments you provided with thoughts on your approach to refactoring the `BlockIcon`. On iOS, I was able to verify that the respective embed blocks that were enabled were shown within the Inserter ( except Instagram which won't be enabled until the Jetpack embed variants are included). My Android Studio is broken so I wasn't able to figure up an Android build to verify the changes there. I will do that tomorrow. 👍🏾

**Comment by fluiddot** [comment]
> Thanks for these changes @fluiddot I reviewed the code and I learnt a lot from the comments you provided with thoughts on your approach to refactoring the `BlockIcon`. 

Thanks @jd-alexander for reviewing the PR ❤️ !

I'd like to note that I recently pushed [a commit](https://github.com/WordPress/gutenberg/pull/34967/commits/ac4553351fd3810d168e5d51f35bc74e37b11424) to solve a conflict, the main change is that the `ALLOWED_EMBED_VARIATIONS` list is now in a different file.

> On iOS, I was able to verify that the respective embed blocks that were enabled were shown within the Inserter ( except Instagram which won't be enabled until the Jetpack embed variants are included). My Android Studio is broken so I wasn't able to figure up an Android build to verify the changes there. I will do that tomorrow. 👍🏾

Ok, I'll wait for your feedback on Android before merging, in my case, I didn't find any issue but better to double-check it 👍, thanks 🙇 !

**Comment by jd-alexander** [comment]
@fluiddot the changes look beautiful on Android. Good job! 🚢

**Comment by fluiddot** [comment]
As it was commented [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3995#issuecomment-930372453), I've just disabled the colored icon versions for the embed blocks in [this commit](https://github.com/WordPress/gutenberg/pull/34967/commits/a155d960d5d52c4769dfd6f87cdc5e2769360ff2).

@jd-alexander let me know if you'd like to do another review due to this change or otherwise, we're safe to merge the PR, thanks 🙇 !

**Comment by jd-alexander** [comment]
> As it was commented [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3995#issuecomment-930372453), I've just disabled the colored icon versions for the embed blocks in [this commit](https://github.com/WordPress/gutenberg/pull/34967/commits/a155d960d5d52c4769dfd6f87cdc5e2769360ff2).
> 
> @jd-alexander let me know if you'd like to do another review due to this change or otherwise, we're safe to merge the PR, thanks 🙇 !

I would just do a quick check to see how it looks on my side! I have never seen the icons rendered without the colors.

**Comment by jd-alexander** [comment]
@fluiddot all is well from my side! Verified that the icons are in unison with the rest that is within the Inserter. LGTM!

**Comment by guarani** [comment]
👋 @fluiddot, I'm testing the Gutenberg Mobile release and had a question about testing instructions. I see the colored icons were removed here (in this [comment](https://github.com/WordPress/gutenberg/pull/34967#issuecomment-930381158)), but I wonder if it should be removed from the PR description. The reason I ask is that when testing the release, to know what to tests I first went to the Gutenberg Mobile PR, then came here and saw mentioned of colored icons. Then when I didn't see colored icons in the app, I came back to the comments here and found it had been removed. 
This same situation might occur when testing the main apps.
I'd understand if you prefer to leave things as-is though, since verifying the colors isn't in the testing instructions, it's instead in the PR description.

**Comment by fluiddot** [comment]
> 👋 @fluiddot, I'm testing the Gutenberg Mobile release and had a question about testing instructions. I see the colored icons were removed here (in this [comment](https://github.com/WordPress/gutenberg/pull/34967#issuecomment-930381158)), but I wonder if it should be removed from the PR description. The reason I ask is that when testing the release, to know what to tests I first went to the Gutenberg Mobile PR, then came here and saw mentioned of colored icons. Then when I didn't see colored icons in the app, I came back to the comments here and found it had been removed. This same situation might occur when testing the main apps. I'd understand if you prefer to leave things as-is though, since verifying the colors isn't in the testing instructions, it's instead in the PR description.

@guarani Thanks for pointing this out 🙇 , I'm afraid I forgot to update that part of the PR description after we removed the colored icons from the changes. I'll review all potential references and update them as soon as possible to prevent any misleading when testing 👍 .

**Pull request merged**
